src/empirical.jl

```diff
@@ -85,16 +85,17 @@ function EmpiricalVariogram(sdata, var₁::Symbol, var₂::Symbol=var₁;
 
   # choose accumulation algorithm
   if algo == :ball && isfloat && isminkowski
-    sums, counts = ball_search_accum(sdata, var₁, var₂, hmax, nlags, distance)
+    sums, cumdists, counts = ball_search_accum(sdata, var₁, var₂, hmax, nlags, distance)
   else
-    sums, counts = full_search_accum(sdata, var₁, var₂, hmax, nlags, distance)
+    sums, cumdists, counts = full_search_accum(sdata, var₁, var₂, hmax, nlags, distance)
   end
 
   # bin (or lag) size
   δh = hmax / nlags
 
   # variogram abscissa
-  abscissa = range(δh/2, stop=hmax - δh/2, length=nlags)
+  abscissa = @. (cumdists / counts)
+  abscissa[counts .== 0] .= range(δh/2, stop=hmax - δh/2, length=nlags)[counts .== 0]
```

Comment:
I wonder if we can simplify this adjustment. Perhaps create the range object right after the `δh`:

```julia
lags = range(...)
```

and then index it here with:

```julia
abscissa[counts .== 0] .= lags[counts .== 0]
```